app/ide-desktop/lib/content/src/index.ts

```diff
@@ -119,104 +119,132 @@ function displayDeprecatedVersionDialog() {
 }
 
 // ========================
-// === Main Entry Point ===
+// === Main entry point ===
 // ========================
 
 interface StringConfig {
     [key: string]: StringConfig | string
 }
 
-class Main {
-    async main(inputConfig: StringConfig) {
-        const config = Object.assign(
-            {
-                loader: {
-                    wasmUrl: 'pkg-opt.wasm',
-                    jsUrl: 'pkg.js',
-                    assetsUrl: 'dynamic-assets',
-                },
-            },
-            inputConfig
-        )
-
-        const appInstance = new app.App({
-            config,
-            configOptions: contentConfig.OPTIONS,
-            packageInfo: {
-                version: BUILD_INFO.version,
-                engineVersion: BUILD_INFO.engineVersion,
+// `new app.App` must be run at least once so that the command line arguments are applied to
+// `configOptions.OPTIONS`.
+let currentAppInstance: app.App | null = new app.App({
+    config: {
+        loader: {
+            wasmUrl: 'pkg-opt.wasm',
+            jsUrl: 'pkg.js',
+            assetsUrl: 'dynamic-assets',
+        },
+    },
+    configOptions: contentConfig.OPTIONS,
+    packageInfo: {
+        version: BUILD_INFO.version,
+        engineVersion: BUILD_INFO.engineVersion,
+    },
+})
+
+function tryStopEnso() {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+    currentAppInstance?.wasm?.drop?.()
+}
+
+async function runEnso(inputConfig?: StringConfig) {
+    tryStopEnso()
+    const rootElement = document.getElementById(IDE_ELEMENT_ID)
+    if (!rootElement) {
+        logger.error(`The root element (the element with ID '${IDE_ELEMENT_ID}') was not found.`)
+    } else {
+        while (rootElement.firstChild) {
```

Comment:
is it still needed at all? I thought we spoke that https://github.com/enso-org/enso/pull/6365 PR will remove `div.scene` with all elements? What other elements we are cleaning here?